File: Homeworks/C#2/StringsAndTextProcessingHomework/15.ReplaceTags/ReplaceTags.cs
/*
Problem 15. Replace tags

Write a program that replaces in a HTML document given as string all the tags <a href="…">…</a> with corresponding tags [URL=…]…/URL].
Example:

input	
<p>Please visit <a href="http://academy.telerik. com">our site</a> to choose a training course. Also visit <a href="www.devbg.org">our forum</a> to discuss the courses.</p>	
 
 output
 <p>Please visit [URL=http://academy.telerik. com]our site[/URL] to choose a training course. Also visit [URL=www.devbg.org]our forum[/URL] to discuss the courses.</p>*/


using System;

namespace RppalceTagsProblem
{
    class ReplaceTags
    {
        static void Main()
        {
            string htmlDocument = "<p>Please visit <a href=\"http://academy.telerik.com\">our site</a> to choose a training course. Also visit <a href=\"www.devbg.org\">our forum</a> to discuss the courses.";

            Console.WriteLine("HTML document:");
            Console.WriteLine(htmlDocument);
            htmlDocument = htmlDocument.Replace("<a href=\"", "[URL=");
            htmlDocument = htmlDocument.Replace("</a>", "[/URL]");
            htmlDocument = htmlDocument.Replace("\">", "]");
            Console.WriteLine();
            Console.WriteLine("Replaced HTML document:");
            Console.WriteLine(htmlDocument);
        }
    }
}

File: Homeworks/JavaScriptFundamentals/08.Strings/08.ReplaceTags/ReplaceTags.js
/*
 Write a JavaScript function that replaces in a HTML document given as string all the tags
 <a href="…">…</a> with corresponding tags [URL=…]…/URL].
 */

var jsConsole,
    button = document.getElementById('replace');

button.onclick = replaceTags;

function replaceTags() {
    var input = jsConsole.read('#input-html'),
        result = '';

    for (var i = 0; i < input.length; i += 1) {
        if (input[i] === '<' && input[i + 1] === 'a') {
            result += '[URL=';
            i += 8;
        } else if (input[i] === '"' && input[i + 1] === '>') {
            result += ']';
            i += 1;
        } else if (input[i] === '<' && input[i + 1]=== '/' && input[i + 2] === 'a') {
            result += '[/URL]';
            i += 3;
        } else {
            result += input[i];
        }
    }

    jsConsole.writeLine(result);
    console.log(result);
}


